### ISSUE TITLE
fish: Unknown command. '<file>' exists but is not an executable file.

### ISSUE THREAD

**Issue by NotCoolKiwi** [fish-shell/fish-shell#9151]
Even when a file/command I execute does NOT exist, I get this error: `fish: Unknown command. '<file>' exists but is not an executable file.` This does not happen with bash or zsh, so I don't believe it is a problem with my `$USER/bin` folder.

Example (everything redacted is my user/hostname, not relevant to the issue):
```
[REDACTED]❙~❱✘≻ thisfiledoesnotexist123                                                                                           ✘ 126
fish: Unknown command. '/home/[REDACTED]/bin/thisfiledoesnotexist123' exists but is not an executable file.
```

Version: 3.5.0
Linux Kernel: 5.18.17-200.fc36.x86_64 (Linux Kernel 5.18.17-200, Fedora 36, x86_64)
Terminals: Tested on cool-retro-term (xterm) and gnome-terminal (xterm-256)

I have tested without customization/config, but it did not help.

**Comment by faho** [comment]
The error message tells us that you have tried to execute something, and we have gone through $PATH and the only thing your system tells us exists is in ~/bin/, but it's not executable or not a file.

What I'm assuming is that our check there is slightly messed up and that we don't get read permissions on ~/bin/.

Which could very well happen on Fedora, what with SElinux and everything.

Please double-check the following:

1. Does ~/bin *exist*?
2. Is it a directory or a symlink?
3. Is it readable, writeable, executable by your user?
4. Are files in your $HOME executable (e.g. is it mounted noexec or are things forbidden via selinux)?


>Version: 3.5.0

I really want to know why distros aren't upgrading to 3.5.1, it's been out for two months with some good fixes. Typically Fedora is alright at this.

**Comment by NotCoolKiwi** [comment]
Thanks for your help! This was not an issue with fish, but with `~/bin`. Sorry for the trouble!

Here is what the issue was:
I had been creating a fish script, and to add it to my ~/bin/ directory, I ran `mv kidasencrtl.sh ~/bin`. However, the bin directory did not exist yet, so the command just renamed `kidasenctrl.sh` to `bin` in my home directory, therefore it was a large file, not a directory. I have now deleted that file and moved my script to `~/.local/bin/`, which is the proper way to do it, and what should have been done from the start. After doing that, this error no longer occurs! For anyone with this issue in the future, [here](https://askubuntu.com/questions/1369412/bash-cd-home-marcoluis-local-bin-not-a-directory) is the stack exchange answer that helped me figure this out. (https://askubuntu.com/questions/1369412/bash-cd-home-marcoluis-local-bin-not-a-directory)
